lib/i18n.rb

```diff
@@ -429,6 +453,17 @@ def normalize_key(key, separator)
           keys
         end
     end
+
+    def interpolation_keys_from_translation(translation)
+      case translation
+      when ::String
+        translation.scan(Regexp.union(I18n.config.interpolation_patterns))
+      when ::Array
+        translation.map { |element| interpolation_keys_from_translation(element) }
```

Comment:
Note that this [does need to be recursive](https://github.com/ruby-i18n/i18n/blob/master/lib/i18n/backend/base.rb#L183-L198)!